Add blogpost about nlp course rename (#2782)
huggingface/blog: llm-course.md

---
title: "The NLP Course is becoming the LLM Course"
thumbnail: /blog/assets/llm-course/llm-course-rename-thumbnail.png
authors:
- user: burtenshaw
- user: reach-vb
- user: lewtun
- user: fdaudens
---

# The NLP Course is becoming the LLM Course!

![image/gif](https://cdn-uploads.huggingface.co/production/uploads/62d648291fa3e4e7ae3fa6e8/I6Pq5TftMqjLOqZzKNplg.gif)

> Hugging Face is (re-)committing to education by expanding and updating its material on [hf.co/learn](http://hf.co/learn). The NLP course has been a cornerstone of Hugging Face and the open source AI community over the last 3 years, but it needs updating and expanding to meet the current needs of the community. 

Reviewer: maybe reword the first line it reads a bit weird

Reviewer: i.e. we've always been committed to education efforts

Reviewer 2: Maybe something like this?

> Education has always been at the heart of Hugging Face’s mission to democratize AI and we’re doubling down on that by giving [hf.co/learn](http://hf.co/learn) a big upgrade! Our NLP course has been a go-to resource for the Hugging Face and open-source AI community for the past 3 years, but it’s time for a refresh. We’re updating and expanding it to keep up with all the exciting stuff happening in AI (which is not easy when there are breakthroughs every week!)

> Education has always been at the heart of Hugging Face’s mission to democratize AI and we’re doubling down on that by giving hf.co/learn a big upgrade! Our NLP course has been a go-to resource for the Hugging Face and open-source AI community for the past 3 years, but it’s time for a refresh. We’re updating and expanding it to keep up with all the exciting stuff happening in AI (which is not easy when there are breakthroughs every week!)

Reviewer 3: ```suggestion
Education has always been at the heart of Hugging Face’s mission to democratize AI and we’re doubling down on that by giving [hf.co/learn](https://hf.co/learn) a big upgrade! Our NLP course has been a go-to resource for the open-source AI community for the past 3 years, and it’s now time for a refresh. We’re updating and expanding it to keep up with all the exciting stuff happening in AI (which is not easy when there are breakthroughs every week!)
```

> We felt this during the successful [agents course](https://huggingface.co/agents-course), where 100k students registered to learn about AI agents in a new, fun, and interactive way! 

Reviewer: also, smol-course?

> We felt this during the experimental [smol-course](https://github.com/huggingface/smol-course) and the massive [agents course](https://huggingface.co/agents-course), where 100k students registered to learn about AI agents in a new, fun, and interactive way! 

Reviewer 3: ```suggestion
We felt the excitement during the experimental [smol-course](https://github.com/huggingface/smol-course) and the massive [agents course](https://huggingface.co/agents-course), where 100k students registered to learn about AI agents in a new, fun, and interactive way! 
```

> Over the last few months we’ve expanded the LLM course with new chapters, including fine-tuning LLMs and building reasoning models like Deepseek R1. These newer chapters don’t fit under the banner of ‘NLP’ so we’ve searched for a more relevant and modern title and landed on **The LLM course.**

Reviewer 3: ```suggestion
Over the last few months we’ve expanded the "NLP" course with new chapters, including fine-tuning LLMs and building reasoning models like Deepseek R1. These newer chapters don’t fit under the banner of ‘NLP’ so we’ve searched for a more relevant and modern title and landed on **The LLM course.**
```

![image/png](https://cdn-uploads.huggingface.co/production/uploads/62d648291fa3e4e7ae3fa6e8/an9U4x7L7rhUEFH_qgYc4.png)

## What’s going to happen to the NLP course material?

We will maintain the existing material that focuses on classic NLP tasks like classification, named entity-recognition and retrieval. These topics are important because:

- We don’t need LLMs for everything!  
- Students still benefit from these simpler tasks that run locally and are easy to interpret.

> In fact, over the coming months we will update and modernise these classic chapters to include approaches like [Sentence Transformers](https://huggingface.co/sentence-transformers), updates in Zero Shot classification, and [ModernBert](https://huggingface.co/blog/modernbert).

Reviewer 4: 🎉

## Will there be new chapters?

Yes. We will add new chapters that focus on making state-of-the-art research accessible to a broader audience and compatible with tools like transformers, spaces, and the Hugging Face Hub. For example, we’ll add chapters on fine-tuning, inference, and retrieval. 

> We will also collaborate with the community to make courses open to users of other libraries. In chapters 11 and 12 we collaborated with libraries and tools to make material and exercises using third party libraries. So far, we’ve worked with:  

Reviewer 4: Consider adding links on `11` and `12` so people can easily go to that content

> We will also collaborate with the community to make courses open to users of other libraries. In chapters 11 and 12 we collaborated with libraries and tools to make material and exercises using third party libraries. So far, we’ve worked with:  

Reviewer 5: ```suggestion
We will also collaborate with the community to make courses open to users of other libraries. In chapters 11 and 12, we collaborated with libraries and tools to make materials and exercises using third-party libraries. So far, we’ve worked with:  
```

> We will also collaborate with the community to make courses open to users of other libraries. In [Chapter 11 on fine-tuning LLMs](https://huggingface.co/learn/nlp-course/en/chapter11/1) and [Chapter 12 on reasoning models](https://huggingface.co/learn/nlp-course/en/chapter12/1), we collaborated with libraries and tools from beyond the Hugging Face ecosystem. So far, we’ve worked with:  

Reviewer 3: This still sounds occasional / exceptional to me. Can we maybe put it more in the frontline?

- [LlamaIndex](https://huggingface.co/llamaindex)  
- [LangChain](https://huggingface.co/langchain-ai)  
- [Unsloth](https://huggingface.co/unsloth)  
- [Marimo](https://huggingface.co/marimo-team)  
- [Maxime Labonne](https://huggingface.co/mlabonne)

> Over the coming months we want to collaborate with more authors, maintainers, and companies so we can build material in the tools you’re using.

Reviewer 4: ```suggestion
Over the coming months, we want to collaborate with more authors, maintainers, and companies so we can prepare material about the tools you’re using.
```
'prepare' can also be 'write', 'build' sounded odd to me. I added a comma to give readers a breather

## Will there be interactive exercises and live sessions?

> Yes, when they’re useful to students. We will focus on reliable written material and coding exercises that can be relied on for years to come. But for the popular topics where students are engaging synchronously, we’ll build interactive exercises and host live sessions.

Reviewer 4: ```suggestion
Yes, when they’re useful to students. We will focus on reliable written material and coding exercises that will remain useful for years to come. But for the popular topics where students are engaging synchronously, we’ll build interactive exercises and host live sessions.
```
I'm avoiding "reliable" and "relied" in the same sentence here

## What’s next?

> If you want to get join the course or think a chapter in the course should be updated with interactive units or host a live session, then follow the [organization on the hub](https://huggingface.co/huggingface-course) and start a discussion item.

Reviewer 4: ```suggestion
To get involved with the course, follow the [organization on the hub](https://huggingface.co/huggingface-course) and start a discussion if you'd like to join, suggest interactive units, or propose a live session.
```
`get join` was incorrect, and `or think ... should be updated ... or host a live session` is a bit odd. I rewrote it from scratch now, but feel free to update this again if you prefer something a bit different.

Author: Thanks. It seems like I gave up before this paragraph. 🤣

> To get involved with the course, follow the [organization on the hub](https://huggingface.co/huggingface-course) and start a discussion if you'd like to join, suggest interactive units, or propose a live session.

Reviewer 3: Should it be renamed too?

Author: Because we can't do much to redirect old links to the org, I'd prefer to leave it as is. But I will rebrand the org and use it much more in comms and the course.

Reviewer 3: Makes sense!
